ccf/app/views.py

```diff
@@ -1,6 +1,17 @@
-from django.shortcuts import redirect
+from django.contrib.auth.decorators import login_required
+from django.contrib.auth.forms import UserCreationForm
+from django.shortcuts import render
+from django.urls import reverse_lazy
+from django.views import generic
 
 
+@login_required
 # Create your views here.
 def homepage_view(request):
-    return redirect("/accounts/login/")
+    return render(request, "tmp_home.html")
+
+
+class SignUp(generic.CreateView):
```

Comment:
Do we need a login required here or have we moved away from that

Reply:
Isn't this where the account signup happens? why would a user need to be logged in to create an account?

Reply:
The `login_required` is necessary as it is used to redirect the user from the base address/homepage to the login page if they are not logged in. It is separate from the sign-up code. I'll add some comments to help explain that.